### ISSUE TITLE
Some Markdown supports does not deal with multibyte character

### ISSUE THREAD

**Issue by Yarakashi-Kikohshi** [microsoft/vscode-markdown-languageservice#17]
This problem still occurs even if you disable all extensions and occurs in both stable and insider.

The following features do not deal with multibyte characters well. It encodes multibyte characters. Do not encode.

- [Markdown path IntelliSense][path-intellisense]
- [Drop into editor to create link][drop-into-editor-to-create-link]
- Markdown rename support
    - [Rename Headers][headers]
    - [Rename Files][files]

The path intellisense for headings works correctly.
Markdown rename support includes “[Rename reference links][reference-links]”, but reference links works correctly.

[drop-into-editor-to-create-link]: https://code.visualstudio.com/updates/v1_67#_markdown-drop-into-editor-to-create-link

[headers]: https://code.visualstudio.com/updates/v1_67#_markdown-rename-headers

[reference-links]: https://code.visualstudio.com/updates/v1_67#_markdown-rename-reference-links

[files]: https://code.visualstudio.com/updates/v1_67#_markdown-rename-files

[path-intellisense]: https://code.visualstudio.com/updates/v1_64#_markdown-path-intellisense

## Demonstrations

```
  .
  |- テスト1.md
  `- テスト2.md
```

```markdown
<!-- テスト1.md -->
# 見出し

[見出し](#見出し)

[Visual Studio Code April 2022][vscode]

[vscode]: https://code.visualstudio.com/updates/v1_67

[テスト2の見出し](テスト2.md#見出し)

```

```markdown
<!-- テスト2.md -->
# 見出し

```

- **Markdown path IntelliSense**

    ![Image path IntelliSense][image_path-intellisense]

- **Drop into editor to create link**

    ![Image Drop into editor to create link][image_drop-into]

- **Rename Headers**

    ![Image Rename headers][image_rename-headers]

    ![Image Rename headers][image_rename-reference-links2]

- **Rename Files**

    ![Image Rename files][image_rename-files]

[image_drop-into]: https://user-images.githubusercontent.com/75878492/170026857-9fa55868-00f7-4f72-b490-0a86b7be4d68.gif

[image_path-intellisense]: https://user-images.githubusercontent.com/75878492/170026837-36034c11-c31d-46d2-86db-29a7f79c5e3f.gif

[image_rename-headers]: https://user-images.githubusercontent.com/75878492/170026843-03e1b117-f8b5-44a1-a9ff-af4ed23c0aaf.gif

[image_rename-files]: https://user-images.githubusercontent.com/75878492/170026854-3ad5e22e-f984-4f02-83e0-4ba3a2cee133.gif

[image_rename-reference-links]: https://user-images.githubusercontent.com/75878492/170026846-a18bf658-6871-45a9-ae23-5344a94c4d3b.gif

[image_rename-reference-links2]: https://user-images.githubusercontent.com/75878492/170026850-9219cb6b-c1f0-42e8-989b-1125612f6136.gif

<details>
<summary>Rename reference links works correctly.</summary>
<div>

![Image Rename reference links][image_rename-reference-links]

</div>
</details>

## Version

```
 Version: 1.67.2 (user setup)
  Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
    Date: 2022-05-17T18:15:52.058Z
Electron: 17.4.1
Chromium: 98.0.4758.141
 Node.js: 16.13.0
      V8: 9.8.177.13-electron.0
      OS: Windows_NT x64 10.0.19043
```

```
 Version: 1.68.0-insider (user setup)
  Commit: c0769274fa136b45799edeccc0d0a2f645b75caf
    Date: 2022-05-23T05:17:12.256Z
Electron: 17.4.4
Chromium: 98.0.4758.141
 Node.js: 16.13.0
      V8: 9.8.177.13-electron.0
      OS: Windows_NT x64 10.0.19043
```

**Comment by mjbvz** [comment]
Please give this another try later once the fix is merged into VS Code 1.80. I fixed a number of different cases here but may have overlooked a few

We should now be more selective about encoding paths rather than always encoding them
